src/data/bedrock/item/ItemSerializerDeserializerRegistrar.php

```diff
@@ -468,6 +470,14 @@ function(Banner $item, int $meta) : void{
 			},
 			fn(Banner $item) => DyeColorIdMap::getInstance()->toInvertedId($item->getColor())
 		);
+		$this->map1to1ItemWithMeta(
+			Ids::FIREWORK_STAR,
+			Items::FIREWORK_STAR(),
+			function(FireworkStar $item, int $meta) : void{
+				$item->getExplosion()->setColors([DyeColorIdMap::getInstance()->fromInvertedId($meta) ?? throw new ItemTypeDeserializeException("Unknown firework star meta $meta")]);
```

Comment:
what happens if the item has colours in the NBT?

Reply:
the color in the meta is actually irrelevant, it is always overwritten with the colors in the NBT. In fact we could silently ignore the meta, its existence would rather seem to be a mojang hack.